src/app/collective-rewards/metrics/context/CycleContext.tsx

```diff
@@ -34,37 +29,36 @@ type CycleProviderProps = {
 }
 
 export const CycleContextProvider: FC<CycleProviderProps> = ({ children }) => {
-  const [timestamp, setTimestamp] = useState(BigInt(DateTime.now().toUnixInteger()))
-
-  useEffect(() => {
-    const interval = setInterval(() => {
-      setTimestamp(BigInt(DateTime.now().toUnixInteger()))
-    }, AVERAGE_BLOCKTIME)
-
-    return () => clearInterval(interval) // Cleanup interval on component unmount
-  }, [])
+  const timestamp = useIntervalTimestamp()
```

Comment:
We need to remove the timestamp from the  `useMemo` dependency.

Reply:
done